tests/unit/parser.js

```diff
@@ -510,10 +512,63 @@ exports.regexp = function (test) {
   TestRun(test).test("var a = 1; var b = a-- / 10;", {esnext: true});
   TestRun(test).test("var a = 1; var b = a-- / 10;", {moz: true});
 
+  TestRun(test)
+    .addError(1, "Invalid regular expression.")
+    .test("var a = /.*/ii;");
+
+  test.done();
+};
+
+exports.regexp.uFlag = function (test) {
+  // Flag validity
+  TestRun(test)
+    .addError(1, "Missing semicolon.")
+    .addError(1, "Expected an assignment or function call and instead saw an expression.")
+    .test("var a = /.*/u;");
+
+  TestRun(test)
+    .test("var a = /.*/u;", { esnext: true });
+
+  // Hexidecimal limits
+  TestRun(test)
+    .addError(3, "Invalid regular expression.")
+    .test([
+      "var a = /\\u{0}/u;",
+      "a = /\\u{10FFFF}/u;",
+      "a = /\\u{110000}/u;"
+    ], { esnext: true });
+
+  // Hexidecimal in range patterns
+  TestRun(test)
+    .addError(3, "Invalid regular expression.")
+    .addError(4, "Invalid regular expression.")
+    .test([
+      "var a = /[\\u{61}-b]/u;",
+      "a = /[\\u{061}-b]/u;",
+      "a = /[\\u{63}-b]/u;",
+      "a = /[\\u{0063}-b]/u;",
+    ], { esnext: true });
+
+  TestRun(test)
+    .test("var x = /[\uD834\uDF06-\uD834\uDF08a-z]/u;", { esnext: true });
+
+  test.done();
+};
+
+exports.regexp.yFlag = function (test) {
+  // Flag validity
+  TestRun(test)
+    .addError(1, "Missing semicolon.")
+    .addError(1, "Expected an assignment or function call and instead saw an expression.")
+    .test("var a = /.*/y;");
```

Comment:
I think it would be better to throw a [W119](https://github.com/jshint/jshint/blob/master/src/messages.js#L197) warning.